src/Type/Php/RegexExpressionHelper.php

```diff
@@ -66,4 +69,59 @@ public function resolve(Expr $expr): Type
 		return $this->initializerExprTypeResolver->getConcatType($concat->left, $concat->right, static fn (Expr $expr): Type => $resolver->resolve($expr));
 	}
 
+	public function getPatternModifiers(string $pattern): ?string
+	{
+		$delimiter = $this->getDelimiterFromString(new ConstantStringType($pattern));
+		if ($delimiter === null) {
+			return null;
+		}
+
+		if ($delimiter === '{') {
```

Comment:
() and [] are also valid reciprocal delimiters.
<> too but that's a shitty one to use really.. Still probably should support for completeness

https://www.php.net/manual/en/regexp.reference.delimiters.php